### ISSUE TITLE
Need a Yaw mode: HEADFREE

### ISSUE THREAD

**Issue by ducminhkt** [TauLabs/TauLabs#1877]
I think we can add a new Yaw mode in TL. In Multiwii it call HeadFree. 
From Multiwii wiki:
"In this mode, the "head" of the multicopter is always pointing to the same direction as when the feature was activated. This means that when the multicopter rotates around the Z axis (yaw), the controls will always respond according the same "head" direction.
With this mode is easier to control the multicopter, even fly it with the physical head towards you since the controls always respond the same. This is a friendly mode to new users of multicopters and can prevent losing the control when you don't know the head direction."
I known AxisLock, but it not really lock, and it not have same behavior like HeadFree.

**Comment by mlyle** [comment]
I agree.  We have some code for loiter mode that is ready for doing command-reprojection.  And it would be nice for attitude mode too.

The real options I see are:

* reproject relative to inverse of home location bearing (like DJI IOC - home lock mode).  IMO only really makes sense in loiter.  Requires a minimum distance from home for position error not to dominate.
* reproject relative to mode activation heading (like "headfree" or DJI IOC - course lock mode)
* reproject relative to arming heading
